[추가] constant file

choichoigang opened the pull request:
**기능구현사항**
- [ o ] ‘지갑화면'에서 동전을 누르면 선택화면에 금액이 표시된다. 지갑화면에서 동전 갯수와 총액도 같이 변경된다.
- [ o ] 투입된 금액이 충족되면 구매할 수 있는 상품의 상품이름 색상이 다른 색으로 변한다.
- [ o ] 구매는 1-9번까지의 선택화면내의 버튼으로 선택해야 한다. 콜라는 1번이고, 칸쵸는 20번이다.
- [ x ] 5초 동안 입력이 없으면 잔액또는 투입된 금액은 반환되고, 지갑에 다시 돈이 채워진다. 
- [ x ] 5초 안에 상품을 선택하면 상품이 나오고, 투입금액을 추가로 넣으면 다시 5초동안 기다린다.
- [ o ] 모든 이벤트는 메시지창 화면에 표시돼야 한다.

**구조**
```bash
├── main.js
├── index.html
├── server.js
├── model
│   ├── vendingMachineModel.js
│   └── walletModel.js
├── view
│   ├── productView.js
│   ├── selectView.js
│   └── walletView.js
├──constant
│   └── vmConstant.js
├── data
│   └── vmData.json
└── styles
    ├── reset.css
    └── style.css
```

choichoigang committed:
[수정] line . 코드 정렬

choichoigang committed:
[추가] constant file

상수값 파일화

choichoigang commented:
pull request 중에 실수가 있어서 다시 올립니다 .... ㅠ

choichoigang committed:
Merge branch 'dev' of https://github.com/codesquad-memeber-2020/vm-1 into feature-js-timer

crongro commented on src/main.js:
```diff
@@ -1,14 +1,16 @@
+
+import VM from "./constant/vmConstant.js";
 
 const _ = {
```
유틸리티도 아닌데 언다바로 이름 짓는건 별로 인거 같아요.
언더바는 보통 아무것도 아닌 유틸리티 묶음에 쓰이는게 보통이라서요.

choichoigang replied:
다른 이름을 고안하도록 하겠습니다 !

crongro reviewed (approved):
커밋이 하나네요?
master로 보내는거 보니 지난주 거 같군요!

choichoigang committed:
[추가] timerModel

setTimeout이 발생할 때 마다 실행되서 잔돈을 반환시켜주는 addEvent 등록
User의 wallet list에서 금액을 투입하는 event가 발행할 때 마다 금액을 투입시켜주는 AccumulatedSum 등록 (WalletView)의 bindEvent method 내부에서 실행

crongro commented:
이건 리뷰 끝났고 close상태를 볼게요.

choichoigang committed:
[추가] timerModel , walletModel , vendingMachineModel 관계 추가

timerModel (walletModel 상속)
VendingMachineModel (timerModel 상속)

choichoigang committed:
[추가] walletView bindEvent method 반환 기능 관련 추가

선택한 wallet List의 count 값이 0일때 return 추가
setTimeout 누적을 방지하기 위해서 event 발생할 때 마다 setTimeout을 초기화 해주는 clearTimeout 추가
timerModel constructor에 state,timer 값에 매번 setTimeout을 할당
productView 와 selectVIew를 초기화 해주는 resetVMModel method 구현 후 호출

choichoigang committed:
[추가] VendingMachineModel 잔돈 반환 로직 추가

<VendingMachineModel >
state :                            change-( productView에 전달될 시 true 라면 productView를 초기화 시켜주는 값 )
                                      type-(select View에 type이 timer로 넘어가는 추가사항 구현)

addEventDial method : switch문으로 변경
                                     number type click 시 clearTimeout 추가
                                     select type click 시 accumulatedAmount 값에 따라서 return 하는 로직 구현 && 잔돈 반환시키는 로직 구현

resetVMModel method : select 시 selectView , productView에 필요한 default 값 설정

choichoigang committed:
[추가] productView , selectView 잔돈 반환 기능 추가

<productView>
change를 인자로 받아서 초기화 상황에 view를 초기화 해주는 기능 추가
 <selectView>
type 값이 timer인 상황을 추가 잔돈 반환시 나오는 Log massage  추가

choichoigang committed:
[삭제] walletModel

alert 기능 삭제

choichoigang committed:
Merge pull request #25 from codesquad-memeber-2020/feature-js-timer

Feature js timer

crongro commented on src/model/timerModel.js:
```diff
@@ -0,0 +1,42 @@
+
+    this.walletValueList = [10000, 5000, 1000, 100, 50, 10];
+  }
+  addEvent() {
```
이름만보고 event를 추가하는(등록하는) 함수인줄 알았는데,
내용은 다른거 같아요.

crongro commented on src/model/timerModel.js:
```diff
@@ -0,0 +1,42 @@
+    this.walletModel.beginningSum();
+    this.walletModel.notify(this.walletModel.walletData);
+
+    return 0;
```
addEvent라는 함수가 0을 반환하는게 어떤 의도인지 알기 어렵습니다.
초기화된 어떤 값이라도, 왜 초기화를 시키는지 모르겠고요.

crongro commented on src/view/productView.js:
```diff
@@ -5,27 +5,33 @@ class ProductView {
+  active(productListEle, productData, change) {
+    if (change) {
+      productData.forEach((ele, idx) => {
+        productListEle.children[idx].className = "";
```
class가 한 개 이상이 될 수도 있으니, 추가/삭제하는 것이 더 명시적일거 같아요.
classList객체의 메서드들을 활용해서요.

crongro commented on src/view/walletView.js:
```diff
@@ -9,12 +9,24 @@ class WalletView {
       this.walletModel.addEvent(e);
-      this.vendingMachineModel.addEventWallet(target.closest('li').querySelector('.count_index').textContent);
+      this.vendingMachineModel.addEventWallet(
+        target.closest("li").querySelector(".count_index").textContent
```
인자로 들어가는 코드가 너무 길어서  보기 어렵네요.
디버깅할때도 불리한 코드 같고요.
변수로 담아두고 사용하시죠.

const content = target.closest("li").querySelector(".count_index").textContent;

crongro commented on src/model/VendingMachineModel.js:
```diff
@@ -21,33 +25,61 @@ class VendingMachineModel extends Observable {
+  resetVMModel() {
+    this.state.accumulatedAmount = 0;
+    this.state.type = "timer";
+    this.state.change = true;
```
change라는 것이 true/false를 갖는 다는 것이 어떤의미인지 모르겠어요.
잔돈이 있냐 없냐라고 한다면,  changeMoney값이 0이냐 그 이상이냐 같은걸로 판단할 수는 없을까요?

crongro commented on src/model/VendingMachineModel.js:
```diff
@@ -21,33 +25,61 @@ class VendingMachineModel extends Observable {
     this.state.targetValue = target.value;
     this.state.accumulatedAmount += Number(target.value);
-    this.state.type = 'wallet';
+    this.state.type = "wallet";
```
type이 wallet 말고 또 뭐가 있죠? 
type개념이 어색해요. type들간의 어떤 공통점도 존재할 거 같거든요. 
wallet, err 이런것들이  type으로써 적절한 것인지 모르겠습니다.

crongro reviewed (approved):
수고하셨어요. 코드를 보니 MVC가 적절히 나눠진 것이라 생각됩니다.
리뷰 참고하시고요.
